runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollService.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2018 Abex
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.cluescrolls;
+
+import com.google.inject.Inject;
+import java.util.List;
+import net.runelite.api.NPC;
+
+public class ClueScrollService
+{
+	@Inject
+	private ClueScrollPlugin plugin;
+
+	public List<NPC> getNpcsToMark()
+	{
+		return plugin.getImmutableNpcsToMark();
```

Comment:
You can just do Collections.immutableList(plugin.getNpcsToMark()) here instead of adding new getter

Reply:
This is to not alloc in a getter.

Reply:
Isnt immutable list just view to original one? should not be very costly and I would personally prefer that small alloc than new field in plugin who do not needs it at all just to avoid it. Or alternatively you can move the pre-allocated field here.

Reply:
IMO any method labed `get` should be a) very cheap to call and b) pure. `Collections.immutableList` is not pure

Reply:
Well then move the field to the ClueScrollService class as it is relevant only here